qiskit_nature/second_q/operators/fermionic_op.py

```diff
@@ -148,6 +148,12 @@ class FermionicOp(SparseLabelOp):
             considered a lower bound, which means that mathematical operations acting on two or more
             operators will result in a new operator with the maximum number of spin orbitals of any
             of the involved operators.
+    .. note::
+
+        A FermionicOp can contain :class:`qiskit.circuit.Parameter` objects as coefficients.
+        However, a FermionicOp containing Parameters does not support the following methods:
+        - ``is_hermitian``
+        - ``to_matrix``
```

Comment:
What does this mean exactly - we don't seem to check if it contains Parameters and raise an error. It just fails in someway - be nice to raise an error with a suitable message about parameters

Reply:
We could check if any coefficients are Parameters before performing the operation. However, that would incur some overhead. I decided to follow the design of SparsePauliOp from Terra and not perform this check. The error message gets raised in a different part of the stack, but the error is still about parameters, so I think it's still helpful enough.

Reply:
Ok this is the error, it doesn't exactly mention parameters, but it's still reasonably helpful.
```
Traceback (most recent call last):
  File "/home/kjs/worktrees/qiskit-nature/qiskit-nature-parameters/test.py", line 7, in <module>
    op.is_hermitian()
  File "/home/kjs/worktrees/qiskit-nature/qiskit-nature-parameters/qiskit_nature/second_q/operators/fermionic_op.py", line 509, in is_hermitian
    return all(np.isclose(coeff, 0.0, atol=atol) for coeff in diff.values())
  File "/home/kjs/worktrees/qiskit-nature/qiskit-nature-parameters/qiskit_nature/second_q/operators/fermionic_op.py", line 509, in <genexpr>
    return all(np.isclose(coeff, 0.0, atol=atol) for coeff in diff.values())
  File "<__array_function__ internals>", line 180, in isclose
  File "/usr/local/venvs/qiskit-nature-parameters-YbHAS_rw/lib/python3.10/site-packages/numpy/core/numeric.py", line 2372, in isclose
    xfin = isfinite(x)
TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

Reply:
In any case what I did is consistent with Terra and also more performant.

Reply:
I guess I might be inclined to sacrifice a little performance for a check to raise a more meaningful error.

Reply:
I agree. In the same way that `PolynomialTensor` has two methods `is_sparse` and `is_dense`, we could add a method `is_parameterized` which could check for this.
All methods which do not support parameterized ops should then call said method and raise a more meaningful error than they do now

Reply:
Done.